tests/integ/modin/index/test_index_methods.py

```diff
@@ -21,10 +22,24 @@
 )
 
 
+def to_snow_index(native_index):
+    """
+    Convert native pandas index to snowpark pandas index.
+    Args:
+        native_index: Native pandas index.
+
+    Returns:
+        Snowpark pandas index.
+    """
+    if isinstance(native_index, native_pd.DatetimeIndex):
+        return pd.DatetimeIndex(native_index)
+    return pd.Index(native_index)
+
+
 @sql_count_checker(query_count=2)
 @pytest.mark.parametrize("native_index", NATIVE_INDEX_TEST_DATA)
 def test_index_copy(native_index):
-    snow_index = pd.Index(native_index)
+    snow_index = to_snow_index(native_index)
```

Comment:
Can you double check `pd.Index(pd.DatetimeIndex([123]))` should still return a DatetimeIndex? 
Because `native_pd.Index(native_pd.DatetimeIndex([123]))` return DatetimeIndex.

So without any change, this test should work too.

Reply:
Added test.